### PR TITLE
Fix/plugin document setting panel name

### LINKED ISSUE #22049: Issue registering a PluginDocumentSettingPanel plugin
I was having trouble to set the settings panel to opened when registering a plugin. Here's the example code:

```javascript
  const NodeDocumentSettings = () => {
    return (
      <PluginDocumentSettingPanel
        className="node-document-settings"
        title="Node"
        opened={true}
      >
        <div>Node settings</div>
      </PluginDocumentSettingPanel>
    );
  };

  registerPlugin('node-document-settings', {
    render: NodeDocumentSettings,
  });
```

As you see on the code, I tried to use the `opened` prop but to no avail.
So I tried to dispatch `toggleEditorPanelOpened('node-document-settings')` but that also didn't work.
I found out that the plugin is registered at local storage like this:
`node-document-settings/undefined`
So using `toggleEditorPanelOpened('node-document-settings/undefined')` worked.

Checked the code at https://github.com/WordPress/gutenberg/blob/v8.0.0/packages/edit-post/src/components/sidebar/plugin-document-setting-panel/index.js#L109
and it seems the component sets the plugin name joining the context name and props name.
I don't think it makes sense to "join" both plugin context name and prop name at the PluginDocumentSettingPanel component. Maybe it was supposed to have `panelName: context.name,` ?

This is happening using Drupal Gutenberg with Gutenberg v8.0.0 but I believe it can be reproduced on WordPress.

### PR DESCRIPTION
**Pull request by ryanwelcher**
## Description
This PR is a first pass at close #22049 

## Types of changes
Instead of combining the context.name and potentially missing ownProps.name. I have update the logic to use context.name only if ownProps.name is not defined. This will address the issue of the panel being referenced as `plugin-slug/undefined` if there is no `name` property passed to the `PluginDocumentSettingPanel` component.

## Checklist:
- [x] My code is tested.
- [x] My code follows the WordPress code style. 
- [x] My code follows the accessibility standards. 
- [x] My code has proper inline documentation. 
- [x] I've included developer documentation if appropriate. 
- [x] I've updated all React Native files affected by any refactorings/renamings in this PR.

### CODE REVIEW

File: docs/designers-developers/developers/slotfills/plugin-document-setting-panel.md [hunk 1]
```diff
@@ -30,3 +30,13 @@ registerPlugin( 'plugin-document-setting-panel-demo', {
 	icon: 'palmtree',
 } );
 ```
+## Accessing a panel programmatically
+
+Custom panels are namespaced with the plugin name that was passed to `registerPlugin`.
+In order to access the panels using function such as `wp.data.dispatch('core/edit-post').toggleEditorPanelOpened` or wp.data.dispatch('core/edit-post').toggleEditorPanelToggle` be sure to prepend the namepace.
```
Comment (gziolo) [inline docs/designers-developers/developers/slotfills/plugin-document-setting-panel.md:36]:
```suggestion
In order to access the panels using function such as `wp.data.dispatch( 'core/edit-post' ).toggleEditorPanelOpened` or `wp.data.dispatch( 'core/edit-post' ).toggleEditorPanelToggle` be sure to prepend the namepace.
```

File: docs/designers-developers/developers/slotfills/plugin-document-setting-panel.md [hunk 2]
```diff
@@ -30,3 +30,13 @@ registerPlugin( 'plugin-document-setting-panel-demo', {
...
+
+To programmatically toggle the custom panel added in the example above, use the following:
+
+```js
+wp.data.dispatch('core/edit-post').toggleEditorPanelToggle('plugin-document-setting-panel-demo/custom-panel');
```
Comment (gziolo) [inline docs/designers-developers/developers/slotfills/plugin-document-setting-panel.md:41]:
```suggestion
wp.data.dispatch( 'core/edit-post' ).toggleEditorPanelToggle( 'plugin-document-setting-panel-demo/custom-panel' );
```

File: packages/block-editor/src/components/colors/test/utils.js [hunk 3]
```diff
@@ -1,118 +0,0 @@
-/**
```
Comment (gziolo) [inline packages/block-editor/src/components/colors/test/utils.js:1]: It doesn't look like this change is intentional.

**Review by gziolo — COMMENTED** [review]
It looks good, thank you for proposing this improvement. 

There is one file that was removed, I assume by accident. Otherwise, it looks good and should be good to go once my code style changes are applied to conform with WordPress guidelines.

File: packages/block-editor/src/components/colors/test/utils.js (hunk 3 above)
Comment (ryanwelcher) [reply packages/block-editor/src/components/colors/test/utils.js]: Thanks for catching that! I've readded the file from `master`

**Review by gziolo — APPROVED** [review]
Cool, thank you for quick improvements :)

**Comment by ryanwelcher** [comment]
@gziolo I found a couple of small items that I addressed.

**Pull request merged**